crates/parser/src/grammar/expressions.rs

```diff
@@ -562,7 +562,12 @@ fn arg_list(p: &mut Parser) {
         if !expr_with_attrs(p) {
             break;
         }
-        if !p.at(T![')']) && !p.expect(T![,]) {
+        if !p.at(T![')']) && !p.eat(T![,]) {
+            // test_err arg_recover
+            // fn main() {
+            //     foo(y, x:, z)
+            // }
+            p.err_recover(&format!("expected {:?}", T![,]), TokenSet::EMPTY);
```

Comment:
In this case we do not want to break out of the loop I think.
The way this stands now is that `foo(y, x:, z)` will stop the arg list after the `:`, I think it would be better to eat the `:` and recover before the branch here and then continue on as is. That way the arg list will be fully parsed.

Reply:
In other cases like below, we can't find a specific condition to exit the loop. All the following code will be parsed as arg list which might not be an expected behavior.

```rust
fn main() {
     foo(y, x
}

fn bar() {
}
```

Reply:
If we only recover(and continue the loop) on `:` and otherwise don't change anything that shouldn't happen. Basically I'm thinking of something like:
```rs
    while !p.at(T![')']) && !p.at(EOF) {
        // test arg_with_attr
        // fn main() {
        //     foo(#[attr] 92)
        // }
        if !expr_with_attrs(p) {
            break;
        }
        if p.at(T![':']) {
            // Eat a trailing `:` so the arg-list doesn't break if the user is typing a path
            p.err_and_bump(&format!("expected {:?}", T![,]));
        }
        if !p.at(T![')']) && !p.expect(T![,]) {
            break;
        }
    }
```